packages/std/src/query/wasm.rs

```diff
@@ -88,6 +146,39 @@ impl_hidden_constructor!(
 
 impl QueryResponseType for CodeInfoResponse {}
 
+#[non_exhaustive]
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+pub struct RawRangeResponse {
+    /// The key-value pairs
+    pub data: Vec<RawRangeEntry>,
+    /// `None` if there are no more key-value pairs within the given key range.
+    pub next_key: Option<Binary>,
+}
+
+impl_hidden_constructor!(
+    RawRangeResponse,
+    data: Vec<RawRangeEntry>,
+    next_key: Option<Binary>
+);
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+pub struct RawRangeEntry {
+    // keys renamed to reduce JSON size overhead
```

Comment:
Not sure if it helps or if it is desired but we could also make them JSON pairs:

```json
["iVrWkQ58JFYA4g==","KDo3I4CdTVug+ltn9KK/PuwXs1c="]
```

The size difference will be small

Reply:
Yes, I experimented with that before and I think it works nicely on the Rust side, but gets somewhat ugly on the Go side. There you then have either an `Array[Array[[]byte]]` where the inner `Array` implicitly only has two elements or you need a custom type with custom deserialization (you would still deserialize into `Array[[]byte]` with the same implicit requirement, but at least it's more contained). That's why I went with the normal struct instead, but maybe custom deserialization would not be so bad for that?

Reply:
Just [tried this on the wasmvm side](https://github.com/CosmWasm/wasmvm/commit/c7c60f39cd7bcc3a0f626840beb713e45776f3cb) and it's not looking too bad, so I'll also change to a pair here.